keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/KeelRepository.kt

```diff
@@ -231,4 +270,108 @@ interface KeelRepository {
   ): ArtifactSummaryInEnvironment?
 
   // END ArtifactRepository methods
+
+  fun getResourceEventHistory(id: String, limit: Int = ResourceRepository.DEFAULT_MAX_EVENTS): List<PersistentEvent> {
+    val resource = getResource(id)
+    return resourceEventHistory(id, limit)
+      .let { events ->
+        actuationPauser.addApplicationActuationEvents(events, resource)
+      }
+  }
+
+  fun getResourceStatus(id: String): ResourceStatus {
+    // For the PAUSED status, we look at the `paused` table as opposed to events, since these records
```

Comment:
Calling attention to this comment.

Reply:
This is the only change in logic in this function and the ones just below. Otherwise, they were just moved over from `ResourceRepository`.